ContextualPopupDecorator/ContextualPopupDecorator.js

```diff
@@ -635,7 +648,7 @@ const Decorator = hoc(defaultConfig, (config, Wrapped) => {
 
 			const hasSpottables = Spotlight.getSpottableDescendants(containerId).length > 0;
 			const spotlessSpotlightModal = spotlightRestrict === 'self-only' && !hasSpottables;
-			const shouldSpotPopup = current === activator && direction === this.adjustedDirection && hasSpottables;
+			const shouldSpotPopup = current === activator && direction === this.getFiveWayDirection() && hasSpottables;
```

Comment:
Nice catch!
Why don't we have an object that maps the information instead of calling this function?
e.g.
We have the below object at the top of the file.
```
const PositionToDirection = {
	above: 'up',
	below: 'down',
	left: 'left',
	right: 'right'
};
```
And at this line,

```suggestion
			const shouldSpotPopup = current === activator && direction === PositionToDirection[this.adjustedDirection.split(' ')[0]] && hasSpottables;
```

Reply:
It's fixed